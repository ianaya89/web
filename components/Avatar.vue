<template>
  <app-tooltip v-if="alt!==null" :position="tooltipPosition" :color="tooltipColor">
    <template v-slot:activator>
      <div :class="classes" class="bg-transparent p-px">
        <div :style="{ 'background-image': imageUrl }" :class="[roundedClass, avatarClasses]" class="h-full bg-center bg-no-repeat bg-cover border border-grey-l4" />
      </div>
    </template>
    {{ alt }}
  </app-tooltip>
  <div v-else :class="classes" class="inline-block bg-transparent p-px">
    <div :style="{ 'background-image': imageUrl }" :class="[roundedClass, avatarClasses]" class="h-full bg-center bg-no-repeat bg-cover relative border border-grey-l4">
      <div v-if="hasIcon" class="bottom-0 right-0 absolute z-10 icon-sign -m-1" :class="iconSign">
        <span :class="iconSign" />
      </div>
    </div>
  </div>
</template>

<script>
import AppTooltip from '@/components/AppTooltip'

export default {
  name: 'Avatar',

  components: {
    AppTooltip
  },

  props: {
    alt: {
      type: String,
      default: null
    },
    tooltipPosition: {
      type: String,
      default: 'top'
    },
    tooltipColor: {
      type: String,
      default: 'light'
    },
    src: {
      type: String,
      default: require('../assets/images/thumb-fallback.svg')
    },
    iconSign: {
      type: String,
      default: null
    },
    rounded: {
      type: Boolean,
      default: true
    },
    classes: {
      type: String,
      default: 'h-16 w-16'
    },
    avatarClasses: {
      type: String,
      default: ''
    }
  },

  computed: {
    imageUrl() {
      return `url(${this.src ||
        require('../assets/images/thumb-fallback.svg')})`
    },

    hasIcon() {
      return this.iconSign !== null && this.iconSign !== ''
    },

    roundedClass() {
      return this.rounded ? 'rounded-full' : ''
    }
  }
}
</script>

<style lang="postcss" scoped>
.icon-sign {
  @apply .text-p3 .text-white .text-center .p-1 .w-6 .h-6 .shadow .flex;

  span {
    @apply .m-auto;
  }

  &::before {
    content: none;
  }

  &.fa-facebook-f {
    background-color: #3c5a99;
  }

  &.fa-instagram {
    @apply .text-p2;
    background-color: #dc4175;
  }

  &.fa-youtube {
    background-color: #c4302b;
  }

  &.fa-twitter {
    background-color: #00aced;
  }

  &.fa-twitter-square {
    background-color: #00aced;
  }

  &.fa-tiktok {
    background-color: #000;
  }

  &.fa-weibo {
    background-color: #d52a2c;
  }

  &.fa-globe {
    background-color: #fff;
    color: #999;
  }
}
</style>
