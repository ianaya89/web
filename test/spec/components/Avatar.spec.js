import { shallowMount } from '@vue/test-utils'
import Avatar from '@/components/Avatar'

describe('Avatar', () => {
  it('renders default avatar placeholder', () => {
    const wrapper = shallowMount(Avatar)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders avatar specific size', () => {
    const wrapper = shallowMount(Avatar, {
      propsData: {
        classes: 'h-10 w-10'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders not rounded avatar', () => {
    const wrapper = shallowMount(Avatar, {
      propsData: {
        rounded: false
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders avatar with icon sign', () => {
    const wrapper = shallowMount(Avatar, {
      propsData: {
        iconSign: 'fab fa-facebook-f'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders avatar with specific image', () => {
    const wrapper = shallowMount(Avatar, {
      propsData: {
        src: 'http://someurl.com/image.png'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
