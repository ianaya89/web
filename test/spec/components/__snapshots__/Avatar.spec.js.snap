// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders avatar specific size 1`] = `
<div
  class="inline-block bg-transparent p-px h-10 w-10"
>
  <div
    class="h-full bg-center bg-no-repeat bg-cover relative border border-grey-l4 rounded-full"
    style="background-image: url(thumb-fallback.svg);"
  >
    <!---->
  </div>
</div>
`;

exports[`Avatar renders avatar with icon sign 1`] = `
<div
  class="inline-block bg-transparent p-px h-16 w-16"
>
  <div
    class="h-full bg-center bg-no-repeat bg-cover relative border border-grey-l4 rounded-full"
    style="background-image: url(thumb-fallback.svg);"
  >
    <div
      class="bottom-0 right-0 absolute z-10 icon-sign -m-1 fab fa-facebook-f"
    >
      <span
        class="fab fa-facebook-f"
      />
    </div>
  </div>
</div>
`;

exports[`Avatar renders avatar with specific image 1`] = `
<div
  class="inline-block bg-transparent p-px h-16 w-16"
>
  <div
    class="h-full bg-center bg-no-repeat bg-cover relative border border-grey-l4 rounded-full"
    style="background-image: url(http://someurl.com/image.png);"
  >
    <!---->
  </div>
</div>
`;

exports[`Avatar renders default avatar placeholder 1`] = `
<div
  class="inline-block bg-transparent p-px h-16 w-16"
>
  <div
    class="h-full bg-center bg-no-repeat bg-cover relative border border-grey-l4 rounded-full"
    style="background-image: url(thumb-fallback.svg);"
  >
    <!---->
  </div>
</div>
`;

exports[`Avatar renders not rounded avatar 1`] = `
<div
  class="inline-block bg-transparent p-px h-16 w-16"
>
  <div
    class="h-full bg-center bg-no-repeat bg-cover relative border border-grey-l4"
    style="background-image: url(thumb-fallback.svg);"
  >
    <!---->
  </div>
</div>
`;
